internal/appsec/internal/intake/client_test.go

```diff
@@ -0,0 +1,245 @@
+				method:       http.MethodPost,
+				body:         "no html & éscaping <",
+				expectedBody: "\"no html & éscaping <\"\n",
+			},
+			{
+				name:      "post /endpoint with body marshaling error",
+				endpoint:  "version/endpoint",
+				method:    http.MethodPost,
+				body:      jsonMarshalError{},
+				wantError: true,
+			},
+		} {
+			tc := tc
+			t.Run(tc.name, func(t *testing.T) {
+				t.Parallel()
+				req, err := c.newRequest(tc.method, tc.endpoint, tc.body)
+
+				if tc.wantError {
+					require.Error(t, err)
+					return
+				}
+
+				require.NoError(t, err)
+				require.Equal(t, tc.method, req.Method)
+				require.Equal(t, baseURL+tc.endpoint, req.URL.String())
+				if tc.expectedBody != "" {
+					body, err := ioutil.ReadAll(req.Body)
+					require.NoError(t, err)
+					require.Equal(t, tc.expectedBody, string(body))
+				}
+			})
+		}
+	})
+
+	t.Run("do", func(t *testing.T) {
+		for _, tc := range []struct {
+			name             string
+			reqBody          interface{}
+			expectedReqBody  string
+			respBody         string
+			expectedRespBody interface{}
+			wantError        bool
+			status           int
+		}{
+			{
+				name: "no request nor response bodies",
+			},
+			{
+				name:            "request body without response body",
+				reqBody:         "string",
+				expectedReqBody: "\"string\"\n",
+			},
+			{
+				name:             "request and response body",
+				reqBody:          "request",
+				expectedReqBody:  "\"request\"\n",
+				respBody:         "\"response\"\n",
+				expectedRespBody: "response",
+			},
+			{
+				name:             "no request body and response body",
+				respBody:         "\"response\"\n",
+				expectedRespBody: "response",
+			},
+			{
+				name:      "bad response json",
+				respBody:  "\"oops",
+				wantError: true,
+			},
+			{
+				name:      "error status code",
+				status:    http.StatusUnprocessableEntity,
+				wantError: true,
+			},
+			{
+				name:   "ok status code",
+				status: 200,
+			},
+		} {
+			tc := tc
+			t.Run(tc.name, func(t *testing.T) {
+				srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+					if tc.status != 0 {
+						w.WriteHeader(tc.status)
+					}
+					if tc.expectedReqBody != "" {
+						require.Equal(t, "application/json", r.Header.Get("Content-Type"))
+						body, err := ioutil.ReadAll(r.Body)
+						require.NoError(t, err)
+						require.Equal(t, tc.expectedReqBody, string(body))
+					}
+					_, _ = w.Write([]byte(tc.respBody))
+				}))
+				defer srv.Close()
+
+				c, err := NewClient(srv.Client(), srv.URL)
+
+				req, err := c.newRequest("GET", "endpoint", tc.reqBody)
+				require.NoError(t, err)
+
+				var respBody interface{}
+				err = c.do(context.Background(), req, &respBody)
+				if tc.wantError {
+					require.Error(t, err)
+					return
+				}
+				require.NoError(t, err)
+				require.Equal(t, tc.expectedRespBody, respBody)
+			})
+		}
+	})
+
+	t.Run("do with context", func(t *testing.T) {
```

Comment:
Can we use one-word test names? THat's what we do everywhere. It's a small nit, but nice to be consistent. I wrote about it in our style guide too.

Reply:
Will do.
Thanks for adding it to the style guide.
We should lint that out at some point ;-)